### ISSUE TITLE
Calling mysql update several times in a row crashes.

### ISSUE THREAD

**Issue by Phillaf** [swoole/swoole-src#4623]
Please answer these questions before submitting your issue.

1. What did you do? If possible, provide a simple script for reproducing the error.

I'm hitting the page from my browser. When I hit it once it works, but if I start refreshing the page a few times in a row, my browser shows failed requests. I might be doing something wrong, but I can find a reference in the docs on how to do this.

```
#!/usr/bin/env php
<?php
declare(strict_types=1);

use Swoole\Http\Request;
use Swoole\Http\Response;

\Swoole\Runtime::enableCoroutine(true, SWOOLE_HOOK_ALL);

$http = new Swoole\Http\Server("0.0.0.0", 80);

$config = (new \Swoole\Database\MysqliConfig())
    ->withHost("mysql")
    ->withDbName("user");
$pool = new \Swoole\Database\MysqliPool($config, 64);

$http->on("request", function (Request $request, Response $response) use ($pool) {

    $mysqli = $pool->get();
    $stmt = $mysqli->prepare("UPDATE users SET name = ? WHERE id = ?");

    $id = 1;
    $name = "phil";
    $stmt->bind_param("si", $name, $id);
    $stmt->execute();

    $pool->put($mysqli);

    $response->write("success");
    $response->end();
    return true;
});

$http->start();

```

2. What did you expect to see?
"Success" printed in my browser, and 200 status code for the page load.

3. What did you see instead?
If I wait several seconds between page loads, then it behaves as expecte. However, if I refresh quickly a few times in a row, then I get: "This site can't be reached" in my browser, and the error shows (failed): net:ERR_CONNECTION_RESET or net:ERR_SOCKET_NOT_CONNECTED.

The swoole log look like this
```
swoole_1  | [2022-02-24 23:20:48 #207.4]	INFO	Server is shutdown now
swoole_1  | swoole: signalled
swoole_1  | 2022-02-24 23:20:48,653 INFO exited: swoole (exit status 0; expected)
swoole_1  | 2022-02-24 23:20:49,656 INFO spawned: 'swoole' with pid 225
swoole_1  | 2022-02-24 23:20:50,658 INFO success: swoole entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)

```

5. What version of Swoole are you using (show your `php --ri swoole`)?
```
swoole

Swoole => enabled
Author => Swoole Team <team@swoole.com>
Version => 4.8.1
Built => Oct 29 2021 15:37:43
coroutine => enabled with boost asm context
epoll => enabled
eventfd => enabled
signalfd => enabled
cpu_affinity => enabled
spinlock => enabled
rwlock => enabled
sockets => enabled
openssl => OpenSSL 1.1.1k  25 Mar 2021
dtls => enabled
http2 => enabled
json => enabled
curl-native => enabled
zlib => 1.2.11
mutex_timedlock => enabled
pthread_barrier => enabled
futex => enabled
mysqlnd => enabled
async_redis => enabled

Directive => Local Value => Master Value
swoole.enable_coroutine => On => On
swoole.enable_library => On => On
swoole.enable_preemptive_scheduler => Off => Off
swoole.display_errors => On => On
swoole.use_shortname => On => On
swoole.unixsock_buffer_size => 8388608 => 8388608
```

6. What is your machine environment used (show your `uname -a` & `php -v` & `gcc -v`) ?

```
uname -a
Linux d5fb65451831 5.14.18-300.fc35.x86_64 #1 SMP Fri Nov 12 16:43:17 UTC 2021 x86_64 GNU/Linux

php -v
PHP 8.0.12 (cli) (built: Oct 21 2021 15:24:13) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.12, Copyright (c) Zend Technologies

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/10/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 10.2.1-6' --with-bugurl=file:///usr/share/doc/gcc-10/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-10 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-10-Km9U7s/gcc-10-10.2.1/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-10-Km9U7s/gcc-10-10.2.1/debian/tmp-gcn/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-mutex
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.1 20210110 (Debian 10.2.1-6) 
```

**Comment by NathanFreeman** [comment]
Hi.
I can not reproduce it.
I guess that it is not swoole error.
What error message are you getting from the console?

**Comment by Phillaf** [comment]
I uploaded a test on docker including all of the configs and steps to reproduce here: https://github.com/Phillaf/swoole-mysql-crash

I made 2 interesting discoveries. 
1. If I loop curls at 1second interval, it shuts down a few times, but if I loop the as fast as I can, then it only crashes once at the end.
2. If I don't mount the mysql data dir on my host, then it never crash.

Is it possible that the folder mount is slowing mysql, and then some form of cleanup/destruct of mysqli happens before mysql is done, which in turn causes a swoole server shutdown?
